pkg/scheduler/conf/scheduler_conf.go

```diff
@@ -36,7 +36,7 @@ type Configuration struct {
 	// Name is name of action
 	Name string `yaml:"name"`
 	// Arguments defines the different arguments that can be given to specified action
-	Arguments map[string]string `yaml:"arguments"`
+	Arguments map[string]interface{} `yaml:"arguments"`
```

Comment:
We have to consider how to keep compatible with user developed plugins.

Reply:
> We have to consider how to keep compatible with user developed plugins.

This update will not influence users who make use of the built-in plugins. A little work to match the interface is needed for users who implement custom plugins with arguments. I've no idea how to support that gracefully. Hope to see some advices.

Reply:
Wait until go1.18, we can try to use go's generics

Reply:
> Wait until go1.18, we can try to use go's generics

@william-wang means the update from `string` to `interface` will introduces some code modifications in custom plugins, for example, the value should be converted to `string` with assert after getting the value from `Arguments`. I agree with that finding a way with no code modification for users is the best. If we cannot achieve that, the update is also acceptable for two factors: 
1. Users who make use of custom plugins are few as to the feedback from issues. 
2. The modification is of little work.